public/app/features/visualization/data-hover/DataHoverView.tsx

```diff
@@ -80,14 +79,11 @@ export function getDisplayValuesAndLinks(
       });
     }
 
-    // Sanitize field by removing hovered property to fix unique display name issue
-    const { hovered, ...sanitizedField } = field;
-
     displayValues.push({
-      name: getFieldDisplayName(sanitizedField, data),
+      name: getFieldDisplayName(field, data),
       value,
       valueString: formattedValueToString(fieldDisplay),
-      highlight: field.hovered,
+      highlight: field === hovered,
```

Comment:
Nice! this seems to fix the issue for geomap... but [a version of the barchart issue is back](https://github.com/grafana/grafana/pull/67872) 😬 


https://github.com/grafana/grafana/assets/22381771/1d5b55f7-aeed-4b46-9e6f-2e8ec60b7456

Repro dashboard
[Datalink issue bar chart 2-1706246105092.json](https://github.com/grafana/grafana/files/14060541/Datalink.issue.bar.chart.2-1706246105092.json)

Reply:
we probably need to either avoid copying fields, or make sure that `field.state` cache is populated and also copied to the field copies, so the same name gets used instead of being improperly re-generated.

Reply:
resolved via https://github.com/grafana/grafana/pull/81323/commits/ee8fe4f7e13632b0f648825b87dfea219603eea3